lib/EnsEMBL/REST/Model/Regulatory.pm

```diff
@@ -35,28 +35,174 @@ sub build_per_context_instance {
   return $self->new({ context => $c, %$self, @args });
 }
 
-sub fetch_regulatory {
-  my $self    = shift;
-  my $regf_id = shift ||
-   Catalyst::Exception->throw("No regulatory feature stable ID given. Please specify an stable ID to retrieve from this service");
+sub fetch_regulatory_feature {
+  my ($self, $rf_id) = @_;
 
   my $c          = $self->context;
   my $species    = $c->stash->{species};
 
   my $rf_a = $c->model('Registry')->get_adaptor($species, 'funcgen', 'RegulatoryFeature');
-  my $rf = $rf_a->fetch_by_stable_id($regf_id);
+  my $rf = $rf_a->fetch_by_stable_id($rf_id);
   if(! defined $rf) {
-    Catalyst::Exception->throw("$regf_id not found for $species");
+    Catalyst::Exception->throw("$rf_id not found for $species");
+  }
+  my $result = $rf->summary_as_hash();
+
+  my $activity = $c->request->param('activity');
+  if($activity == 1) {
+    my $data; 
+    for my $ra(@{$rf->regulatory_activity}) {
+      $data->{$ra->epigenome->production_name} = $ra->activity;
```

Comment:
Braveheart shall be my name! On a more serious note: Yes, those are required values in the DB.